src/finn/transformation/streamline/reorder.py

```diff
@@ -93,13 +93,27 @@ def apply(self, model):
                     graph.node.insert(node_ind + 1, new_add)
                     # replace add value
                     model.set_initializer(add_weight_name, BA)
+                    # Delete the datatype annotation of the parameter tensor
```

Comment:
I am not sure that it's necessary to set the datatype and shapes to None. The InferShapes and InferDataTypes transforms should overwrite if necessary. Did you experience an issue when not setting datatype and shape to None?